File: src/main/resources/projectTemplates/java-maven/src/main/java/my/example/Greeter.java
package my.example;

import dev.restate.sdk.Context;
import dev.restate.sdk.annotation.Handler;
import dev.restate.sdk.annotation.Service;
import dev.restate.sdk.endpoint.Endpoint;
import dev.restate.sdk.http.vertx.RestateHttpServer;

@Service
public class Greeter {

  @Handler
  public String greet(Context ctx, String name) {
    return "You said hi to " + name + "!";
  }

  public static void main(String[] args) {
    RestateHttpServer.listen(Endpoint.bind(new Greeter()));
  }
}

File: src/main/resources/projectTemplates/kotlin/src/main/kotlin/my/example/Greeter.kt
package my.example

import dev.restate.sdk.annotation.*
import dev.restate.sdk.http.vertx.RestateHttpServer
import dev.restate.sdk.kotlin.*
import dev.restate.sdk.kotlin.endpoint.*

@Service
class Greeter {

  @Handler
  suspend fun greet(ctx: Context, name: String): String {
    return "You said hi to $name!";
  }
}

fun main() {
  RestateHttpServer.listen(endpoint {
    bind(Greeter())
  })
}